activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -43,65 +50,181 @@ class InvalidMessage < StandardError; end
     # * <tt>:digest</tt> - String of digest to use for signing. Default is +SHA1+.
     # * <tt>:serializer</tt> - Object serializer to use. Default is +Marshal+.
     def initialize(secret, *signature_key_or_options)
-      options = signature_key_or_options.extract_options!
-      sign_secret = signature_key_or_options.first
+      @options = signature_key_or_options.extract_options!
       @secret = secret
-      @sign_secret = sign_secret
-      @cipher = options[:cipher] || 'aes-256-cbc'
-      @verifier = MessageVerifier.new(@sign_secret || @secret, digest: options[:digest] || 'SHA1', serializer: NullSerializer)
-      @serializer = options[:serializer] || Marshal
+      @sign_secret = signature_key_or_options.first || @secret
+      @alg = @options[:alg] || 'dir'
+      @enc = @options[:enc] || 'A128GCM'
```

Comment:
Yeah, you are right. It would be better for this to be 256 bit just like previously.